console/v1/types_console_plugin.go

```diff
@@ -0,0 +1,182 @@
+	// +optional
+	Proxy []ConsolePluginProxy `json:"proxy,omitempty"`
+	// i18n is the configuration of plugin's localization resources.
+	// +kubebuilder:validation:Required
+	// +required
+	I18n ConsolePluginI18n `json:"i18n,omitempty"`
+}
+
+// ConsolePluginI18n holds information on localization resources that are served by
+// the dynamic plugin.
+type ConsolePluginI18n struct {
+	// enabled indicates if the plugin contains localization resource. If enabled,
+	// i18n namespace, that the plugin contains, will be loaded by the console server.
+	// If i18n is not enabled, no i18n namespaces will be loaded for the plugin.
+	// +kubebuilder:validation:Required
+	// +required
+	Enabled bool `json:"enabled,omitempty"`
+}
+
+// ConsolePluginProxy holds information on various service types
+// to which console's backend will proxy the plugin's requests.
+type ConsolePluginProxy struct {
+	// type is the type of the console plugin's proxy. Currently only "Service" is supported.
+	// +kubebuilder:validation:Required
+	// +required
+	Type ConsolePluginProxyType `json:"type"`
+	// alias is a proxy name that identifies the plugin's proxy. An alias name
+	// should be unique per plugin. The console backend exposes following
+	// proxy endpoint:
+	//
+	// /api/proxy/plugin/<plugin-name>/<proxy-alias>/<request-path>?<optional-query-parameters>
+	//
+	// Request example path:
+	//
+	// /api/proxy/plugin/acm/search/pods?namespace=openshift-apiserver
+	//
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=128
+	// +kubebuilder:validation:Pattern=`^[A-Za-z0-9-_]+$`
+	// +required
+	Alias string `json:"alias"`
+	// service is an in-cluster Service that the plugin will connect to.
+	// The Service must use HTTPS. The console backend exposes an endpoint
+	// in order to proxy communication between the plugin and the Service.
+	// Note: service field is required for now, since currently only "Service"
+	// type is supported.
+	// +kubebuilder:validation:Required
+	// +required
+	Service ConsolePluginProxyServiceConfig `json:"service,omitempty"`
+	// caCertificate provides the cert authority certificate contents,
+	// in case the proxied Service is using custom service CA.
+	// By default, the service CA bundle provided by the service-ca operator is used.
+	// +kubebuilder:validation:Pattern=`^-----BEGIN CERTIFICATE-----([\s\S]*)-----END CERTIFICATE-----\s?$`
+	// +kubebuilder:validation:Optional
+	// +optional
+	CACertificate string `json:"caCertificate,omitempty"`
+	// authorize indicates if the proxied request should contain the logged-in user's
+	// OpenShift access token in the "Authorization" request header. For example:
+	//
+	// Authorization: Bearer sha256~kV46hPnEYhCWFnB85r5NrprAxggzgb6GOeLbgcKNsH0
+	//
+	// By default the access token is not part of the proxied request.
+	// +kubebuilder:validation:Required
+	// +required
+	Authorize bool `json:"authorize,omitempty"`
```

Comment:
Bools aren't allowed by convention, see my previous comment

Is this actually required? You have an omitempty which would hide the value if it's set to false

Reply:
right, so this value servers only to inform the console server if the proxied request should contain user token. Its pretty straight forward and not sure how this could be modelled as Enum